### PR TITLE
Fix STU3 validation

### PR DESCRIPTION
**Pull request by SRiviere**
find correct extension profile, use custom datatype profile, allow to disable terminology checks

### CODE REVIEW

File: hapi-fhir-structures-dstu3/src/main/java/org/hl7/fhir/dstu3/elementmodel/Property.java [hunk 1]
```diff
@@ -203,7 +204,7 @@ public boolean isChoice() {
     ElementDefinition ed = definition;
     StructureDefinition sd = structure;
     List<ElementDefinition> children = ProfileUtilities.getChildMap(sd, ed);
-    if (children.isEmpty()) {
+    if (children.isEmpty()  || isElementWithOnlyExtension(ed, children)) {
```
Comment (jamesagnew) [inline hapi-fhir-structures-dstu3/src/main/java/org/hl7/fhir/dstu3/elementmodel/Property.java]: I'm not totally clear on what this is for..

Would you be able to include a unit test that demonstrates what this part is fixing?

**Comment by SRiviere** [comment]
Sure sorry i forget it. In summary is in a resource when you specify a list of possible extension you may have a element like that:

        <element>
            <path value="Patient.extension"/>
            <slicing>
                <discriminator>
                    <type value="value"/>
                    <path value="url"/>
                </discriminator>
                <rules value="open"/>
            </slicing>
            <definition value="May be used to represent additional information that is not part of the basic definition of the element."/>
        </element>

And this element was generating "types == 0, and no children found" when it try to validate an extension

**Comment by SRiviere** [comment]
Sure i'll add the tests. In summary this is to manage a element like that:
<element>
   <path value="Patient.extension"/>
   <slicing>
     <discriminator>
       <type value="value"/>
       <path value="url"/>
     </discriminator>
     <rules value="open"/>
   </slicing>
   <definition value="May be used to represent additional information that is not part of the basic definition of the element."/>
</element>

And avoid a this error: "types == 0, and no children found", because ther is no type define when the extension is defined in is own element so the search should continue

File: hapi-fhir-structures-dstu3/src/main/java/org/hl7/fhir/dstu3/elementmodel/Property.java (hunk 1 above)
Comment (SRiviere) [inline hapi-fhir-structures-dstu3/src/main/java/org/hl7/fhir/dstu3/elementmodel/Property.java]: added

**Comment by jamesagnew** [comment]
Thanks @SRiviere , I've merged this!

**Comment by jamesagnew** [comment]
By the way, I'm crediting this to @SRiviere in the changelog. Please let me know if you would like me to use another name.

**Comment by jamesagnew** [comment]
(And I'm also forward-porting this fix to the R4 validator)

**Pull request merged**